### PR TITLE
feat: add methods and tests for serverVariables

### PR DESCRIPTION
**Pull request by danielkocot**
**Related issue(s)**
asyncapi/spec#707, asyncapi/spec#717, asyncapi/spec-json-schemas#174

### CODE REVIEW

File: test/models/components_test.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const { expect } = require('chai');
+const { it } = require('mocha');
```
Comment (magicmatatjahu) [inline test/models/components_test.js:2]: Why did you need to import `it`? I think we can remove that.

**Review by magicmatatjahu — COMMENTED** [review]
LGTM, however we have to wait for new minor release to merge it, so I will label it as `do not merge` . Thanks!

File: test/models/components_test.js (hunk 1 above)
Comment (danielkocot) [reply test/models/components_test.js:2]: You're right.

**Comment by smoya** [comment]
@danielkocot we have now the release branches for `2.4.0`. Would you mind changing the base branch to `2022-04-release` ? Thanks!

**Comment by smoya** [comment]
@derberg can you approve the workflows so they can run? Thanks

2022-04-05 at 10 56 47

**Comment by smoya** [comment]
@magicmatatjahu would you mind checking if you can approve this PR? [You did](https://github.com/asyncapi/parser-js/pull/476) but we had to wait for this release to happen.

**Review by magicmatatjahu — APPROVED** [review]
LGTM! :) @smoya Done.

**Comment by magicmatatjahu** [comment]
@smoya Could we merge it as next prerelease?

**Comment by smoya** [comment]
@danielkocot As you probably noticed, I merged https://github.com/asyncapi/spec-json-schemas/pull/174, which ended up being properly released as new prerelease `v2.14.0-2022-04-release.2`.

The very last step we require from you in this PR is to change `@asyncapi/specs` pointed version to the latest prerelease, which is `v2.14.0-2022-04-release.2` in `package.json` file and run `npm update @asyncapi/specs`.

**Comment by smoya** [comment]
@danielkocot it seems the branch has now some conflicts with `2022-04-release`. Would you mind fixing them? 🙏
